snappy_wrappers/wrappers/bcftools/regions/wrapper.py

```diff
@@ -13,7 +13,16 @@
     bed = config["bcftools"].get("path_bed", "")
 elif "filter_nb" in snakemake.wildcards.keys():
     filter_nb = int(snakemake.wildcards["filter_nb"]) - 1
-    bed = config["filter_list"][filter_nb]["regions"].get("path_bed", "")
+    keywords = config["filter_list"][filter_nb]["regions"].keys()
+    assert len(keywords) == 1
+    keyword = list(keywords)[0]
+    path = config["filter_list"][filter_nb]["regions"][keyword]
+    if keyword == "include":
+        bed = path
+    elif keyword == "exclude" or "path_bed":
+        bed = f"^{path}"
+    else:
+        bed = ""
```

Comment:
1. Does this logic only ever apply to `filter_nb` or would the include/exclude bit also make sense for the other cases?
2. There can only ever be one keyword, correct? In that case, config validation for filter_list/.../regions should catch that (i.e. the `OneFilter…` classes' `check_config()` functions should do that before we even get here, which would also make snakemake fail earlier in case of misconfiguration)
3. In general, I am not a fan of accessing the config in wrappers, because they then rely on very specific configuration layouts (making them less portable and changes to config structure are more difficult to propagate to the wrappers). I'd advise (for the future) doing necessary setup in the modules (`__init__.py`) or Snakefiles and pass them on via snakemake.params, because that is what these values essentially are -- parameters controlling the execution of the wrapper.

Reply:
Postponed to next push...